lib/ecto/migration.ex

```diff
@@ -658,4 +659,17 @@ defmodule Ecto.Migration do
   defp validate_type!(%Reference{} = reference) do
     reference
   end
+
+  def prefix(opts_prefix) when is_nil(opts_prefix) do
```

Comment:
We can make these private if this can be improved to allow the table to be prefixed before being sent into runner:
```elixir
defmacro alter(object, do: block) do
    quote do
      table = %Table{} = unquote(object)
      Runner.start_command({:alter, %{table | prefix: prefix(table.prefix)}})
      unquote(block)
      Runner.end_command
    end
  end
```
any ideas?